### PR TITLE
Elasticsearch:  Add documentation for supported Elasticsearch query transformations

### LINKED ISSUE #31323: Docs: Add documentation for supported Elasticsearch query transformation
Following up on https://github.com/grafana/grafana/pull/31322

**What would you like to be added**:
Documentation on Elasticsearch datasource Prometheus/Loki query transformation feature.

**Why is this needed**:

https://github.com/grafana/grafana/pull/23670 introduced the ability to transform Prometheus/Loki queries to Elasticsearch log queries but this is not currently documented anywhere.

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:

Adds a small paragraph describing  the query transformation feature in Explore

**Which issue(s) this PR fixes**:

Fixes #31323

**Special notes for your reviewer**:

Shall I backport this to v7.5.x? is it needed to update the docs?

### CODE REVIEW

**Review by ivanahuckova — COMMENTED** [review]
I would  remove note `> Only available in Grafana v6.3+.` as it is long time ago. Not sure if we want to have `(BETA)` in the heading. It was in beta 2 years ago, but I don't think it is now. So I would remove that as well.  Other than that 🤩.

**Comment by oddlittlebird** [comment]
Yes, backport if it is applicably to Grafana 7.5.x.

File: docs/sources/datasources/elasticsearch.md [hunk 1]
```diff
@@ -198,6 +198,8 @@ for annotation events.
 Querying and displaying log data from Elasticsearch is available in [Explore]({{< relref "../explore" >}}), and in the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards.
 Select the Elasticsearch data source, and then optionally enter a lucene query to display your logs.
 
+When switching from a Prometheus or Loki datasource in Explore your query will be automatically translated to an Elasticsearch log query with a correct Lucene filter.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/elasticsearch.md:201]:
```suggestion
When switching from a Prometheus or Loki data source in Explore, your query is automatically translated to an Elasticsearch log query with a correct Lucene filter.
```

**Review by achatterjee-grafana — APPROVED** [review]
Added copy edit change.

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Approved by mistake.

**Comment by achatterjee-grafana** [comment]
> I would remove note `> Only available in Grafana v6.3+.` as it is long time ago. Not sure if we want to have `(BETA)` in the heading. It was in beta 2 years ago, but I don't think it is now. So I would remove that as well. Other than that 🤩.

Agree, all we care about are v7.x and above at this point.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Comment by Elfo404** [comment]
will remove the beta notice and merge :)

**Pull request merged**